agora/Agora/Utils.hs

```diff
@@ -390,3 +409,44 @@ psingletonValue = phoistAcyclic $
         outerTup = pcon $ PMap $ psingleton #$ ppairDataBuiltin # pdata sym # pdata innerTup
         res = pcon $ PValue outerTup
      in res
+
+-- | Finds the TxOut of an effect from TxInfo and TxOutRef
+findTxOutByTxOutRef :: Term s (PTxOutRef :--> PTxInfo :--> PMaybe PTxOut)
+findTxOutByTxOutRef = phoistAcyclic $
+  plam $ \txOutRef txInfo ->
+    pmatch (pfindTxInByTxOutRef # txOutRef # txInfo) $ \case
+      PJust ((pfield @"resolved" #) -> txOut) -> pcon $ PJust txOut
+      PNothing -> pcon PNothing
+
+-- | Get script hash from an Address.
+scriptHashFromAddress :: Term s (PAddress :--> PMaybe PValidatorHash)
+scriptHashFromAddress = phoistAcyclic $
+  plam $ \addr ->
+    pmatch (pfromData $ pfield @"credential" # addr) $ \case
+      PScriptCredential h -> pcon $ PJust $ pfield @"_0" # h
+      _ -> pcon PNothing
+
+-- | Find all TxOuts sent to an Address
+findOutputsToAddress :: Term s (PTxInfo :--> PAddress :--> PBuiltinList PTxOut)
+findOutputsToAddress = phoistAcyclic $
+  plam $ \info address' -> P.do
+    address <- plet $ pdata address'
+    let outputs = pfromData $ pfield @"outputs" # info
+        filteredOutputs =
+          pfilter
+            # plam
+              ( \(pfromData -> txOut) -> P.do
+                  selfAddress <- plet $ pfield @"address" # txOut
```

Comment:
Oh I missed one. `pelt` here is redundant😢

Reply:
I already fixed that : )